champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/brute/passives/Juggle.java

```diff
@@ -0,0 +1,272 @@
+        if (UtilBlock.isGrounded(player, 1) || UtilBlock.isInWater(player)) return;
+
+        event.setKnockback(false);
+        event.addReason(getName());
+
+        onHit(player, event.getDamagee(), false);
+    }
+
+    public void onHit(Player player, LivingEntity target, boolean isFriendly) {
+        UUID playerUUID = player.getUniqueId();
+        JuggleData juggleData = data.get(playerUUID);
+
+        if (juggleData == null) return;
+
+        final int curCharges = juggleData.getCharges();
+
+        int level = getLevel(player);
+        if (curCharges >= getMaxCharges(level)) {
+            championsManager.getCooldowns().use(player, getName(), getTimeBetweenCharges(level), false, true, true);
+        }
+
+        final int newCharges = Math.max(0, curCharges - 1);
+        juggleData.setCharges(newCharges);
+        notifyCharges(player, newCharges);
+
+        // Movement
+        Vector upward = new Vector(0, 1, 0);
+
+        VelocityData velocityData;
+        if (isFriendly) {
+            velocityData = new VelocityData(upward, allyVelocity, false, allyYSet, allyYAdd, allyYMax, true);
+
+            taskScheduler.addTask(new BPVPTask(target.getUniqueId(), uuid -> !UtilBlock.isGrounded(uuid), uuid -> {
+                Player allyToApplyNoFallTo = Bukkit.getPlayer(uuid);
+                if (allyToApplyNoFallTo != null) {
+                    championsManager.getEffects().addEffect(allyToApplyNoFallTo, player, EffectTypes.NO_FALL, getName(), (int) fallDamageLimit,
+                            250L, true, true, UtilBlock::isGrounded);
+                }
+            }, 1000));
+        } else {
+            velocityData = new VelocityData(upward, enemyVelocity, false, enemyYSet, enemyYAdd, enemyYMax, true);
+        }
+
+        UtilVelocity.velocity(target, player, velocityData, VelocityType.CUSTOM);
+
+        // No feedback messages but send sound for everyone
+        target.getWorld().playSound(target.getLocation(), Sound.ENTITY_BREEZE_DEFLECT, 2.0F, 0.8F);
+
+
+        // A dark purple dust particle effect
+        var dust = new Particle.DustOptions(Color.fromRGB(128, 0, 128), 1.2f);
+
+        Location base = target.getLocation().add(0, 0.8, 0);
+
+        // Spawn particles in a spiral pattern around the target (more of a cone though)
+        for (double y = 0; y <= 1.5; y += 0.05) {
+            double radius = 0.2 + y * 0.2; // tighter spiral
+            for (int i = 0; i < 3; i++) {
+                double angle = y * 4 + i * Math.PI * 2 / 3;
+                double x = Math.cos(angle) * radius;
+                double z = Math.sin(angle) * radius;
+                base.getWorld().spawnParticle(Particle.DUST, base.clone().add(x, y, z), 0, dust);
+            }
+        }
+    }
+
+
+
+    @Override
+    public void trackPlayer(Player player, Gamer gamer) {
+        data.computeIfAbsent(player.getUniqueId(), k -> new JuggleData());
+    }
+    @UpdateEvent(delay = 250)
+    public void addCharge() {
```

Comment:
Could land a plane in the white space above this function. Should also insert whitespace below this function